packages/@aws-cdk/aws-ec2/lib/cfn-init-elements.ts

```diff
@@ -327,6 +327,9 @@ export abstract class InitFile extends InitElement {
    * Use a literal string as the file content
    */
   public static fromString(fileName: string, content: string, options: InitFileOptions = {}): InitFile {
+    if (!content) {
+      throw new Error('Content was empty');
```

Comment:
This could be more descriptive, at least referencing the file that was trying to be created. Also, we like our error messages to suggest a clear path forward.

How about:

```
InitFile ${fileName}: cannot create empty file. Please supply at least one character of content.
```

Reply:
Good suggestion, fixed.